packages/astro/src/@types/astro.ts

```diff
@@ -1533,9 +1533,12 @@ export interface AstroUserConfig {
 			 *  - `prefix-always`: All URLs will display a language prefix.
 			 *    URLs will be of the form `example.com/[locale]/content/` for every route, including the default language.
 			 *    Localized folders are used for every language, including the default.
+			 *  - `domain`: SSR only, it enables support for different domains. When a locale is mapped to domain, all the URLs won't have the language prefix.
+			 *    You map `fr` to `fr.example.com`, if you want a to have a blog page to look like `fr.example.com/blog` instead of `example.com/fr/blog`.
+			 *    The localised folders be must in the `src/pages/` folder.
```

Comment:
I have some questions about how this works to be sure I understand:

1. if you select `routingStrategy: "domain"`, what happens to the URLs for your `defaultLocale`? Does it depend on whether you're using a `/[defaultLocale]/` folder structure? Does it make any assumptions by default?
2. Is it correct to assume that for non-default locales, which will all be written in `/locale/` folders, that the URLs will be prefixed if you don't specifically add them to `domains`? So, it's possible for a site to have a mixture of e.g. `example.com/fr/` and `es.example.com`?
3. is `domains` *ignored* if `routingStrategy` is set to one of the two other options? Could it cause problems to have `domains` configured if you do NOT have `domain` set as your routing strategy?
4. Does the order of `domains` and `routingStrategy` matter in the `i18n` configuration? Could someone run into trouble if these are not in the proper order?
5. Looking at the example below, `routingStrategy` is outside of the object containing the other config options. [In the docs](https://docs.astro.build/en/guides/internationalization/#configure-i18n-routing), we show it as inside this object. Can you confirm which should be correct?

Thanks!

Reply:
1. Yeah, there are assumptions. If you want to map a locale to a domain, the locale folder **must be** at `src/pages/[locale]`. The same goes for the `defaultLocale`. If the default locale **isn't** mapped to a domain, there aren't any assumptions. Although we are still evaluating the behaviour of `defaultLocale`, so it's possible we will change it soon. 
2. Yes, that is correct.
3. No, that will be an error. For example, if you have `i18n.domains` and `routingStategy: "prefix-always"`, Astro will throw a validation nerror.
4. No, the order doesn't matter.
5. Mmmmh, `routingStategy` is inside the `i18n`, same goes for `domains`. Are you referring to something else?